lib/internal/fs/promises.js

```diff
@@ -678,6 +695,18 @@ async function writeFile(path, data, options) {
   return PromisePrototypeFinally(writeFileHandle(fd, data), fd.close);
 }
 
+function isIterable(obj) {
```

Comment:
The name is a bit unfortunate, maybe its name should reflect the fact that some iterables are ruled out? Maybe `isNonBuiltInIterable`? `isUserIterable`? `isCustomIterable`?